doc/mgmt/SONiC_MGMT_Tacacs_HLD.md

```diff
@@ -0,0 +1,220 @@
+# Feature Name
+TACACS+ Support in Management Framework
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/29/2019  |   Joyas Joseph     | Initial version                   |
+
+# About this Manual
+This document provides general information about TACACS+ support in SONiC Management Framework
+# Scope
+The scope of this document is within the bounds of the functionality provided by the new SONiC Management Framework. The underlying TACACS+
+support in SONiC is provided by this high-level design document:
+[TACACS+ Authentication](https://github.com/Azure/SONiC/blob/master/doc/aaa/TACACS%2B%20Authentication.md)
+
+## Table 1
+
+| Term |     Meaning            |
+|:----:|:----------------------:|
+|NBI   |	North Bound Interface |
+
+
+# 1 Feature Overview
+This feature allows the user to configure TACACS+ Authentication using NBI (CLI/REST/gNMI) provided by SONiC Management Framework. Configuration changes from the user are pushed to CONFIG DB. The implementation is contained within the Management Framework container.
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+
+### 1.1.2 Configuration and Management Requirements
+1. CLI configuration/show support
+2. REST API support
+3. gNMI support
+
+### 1.1.3 Scalability Requirements
+
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+1. Implement TACACS+ support using transformer in sonic-mgmt-framework.
```

Comment:
We are planning to implement the RADIUS server support in sonic-mgmt-framework as well. I'm thinking, that the overlap would be minimal, since the oc-yang paths for RADIUS, and TACACS+ are separate. Is this valid ?

Reply:
The overlap would be in the aaa command.

Reply:
Yes, the server-group would be under aaa and system.
/oc-sys:system/oc-aaa:aaa/oc-aaa:server-groups/oc-aaa:server-group=<somegroupfortacacs+>/servers/server=10.24.1.15/tacacs/config/ 

Is there a translib app for system already ?

Reply:
Yes. There is a translib app for system but that doesnt implement aaa/tacacs etc. We will be migrating the existing one to use transformer capabilities.